File: Patrones/Creacionales/Builder/Java/src/DirectorDeCasa.java
public class DirectorDeCasa {
    private final ConstructorDeCasa constructorDeCasa;

    public DirectorDeCasa(ConstructorDeCasa constructorDeCasa) {
        this.constructorDeCasa = constructorDeCasa;
    }

    public void construirCasaSimple() {
        this.construccionBasica("Simple");
        constructorDeCasa.construirGarage();
    }

    public void construirCasaCompleta() {
        this.construccionBasica("Completa");
        constructorDeCasa.construirGarage();
        constructorDeCasa.construirPiscina();
        constructorDeCasa.construirJardin();
    }

    private void construccionBasica(String nombre)
    {
        constructorDeCasa.crearNuevaCasa(nombre);
        constructorDeCasa.construirCimientos();
        constructorDeCasa.construirEstructura();
        constructorDeCasa.construirTecho();
    }

    public Casa getCasa() {
        return constructorDeCasa.getCasa();
    }
}

File: Patrones/Creacionales/Builder/Python/DirectorDeCasa.py
class DirectorDeCasa:
    def __init__(self, constructor_de_casa):
        self.__constructor_de_casa = constructor_de_casa

    def construir_casa_simple(self):
        self.__construccion_basica("Simple")
        self.__constructor_de_casa.construir_garage()

    def construir_casa_completa(self):
        self.__construccion_basica("Completa")
        self.__constructor_de_casa.construir_garage()
        self.__constructor_de_casa.construir_piscina()
        self.__constructor_de_casa.construir_jardin()

    def __construccion_basica(self, nombre):
        self.__constructor_de_casa.crear_nueva_casa(nombre)
        self.__constructor_de_casa.construir_cimientos()
        self.__constructor_de_casa.construir_estructura()
        self.__constructor_de_casa.construir_techo()

    def get_casa(self):
        return self.__constructor_de_casa.get_casa()